module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp

```diff
@@ -144,97 +150,132 @@ namespace module::behaviour::planning {
-                    * (1.0f / std::nano::den);
-
-
-                Eigen::Vector3d rBWw_temp(ball.position.x(), ball.position.y(), fieldDescription.ball_radius);
-                rBWw     = timeSinceBallSeen < search_timeout ? rBWw_temp :                         // Place last seen
-                           Htw.inverse().linear().leftCols<1>() + Htw.inverse().translation();  // In front of the robot
-                position = (Htw * rBWw).head<2>();
-
-                // Hack Planner:
-                float headingChange = 0;
-                float sideStep      = 0;
-                float speedFactor   = 1;
-                if (useLocalisation) {
-
-                    // Transform kick target to torso space
-                    auto fieldPosition = Eigen::Affine2d(field.position);
-                    Eigen::Affine3d Hfw;
-                    Hfw.translation() =
-                        Eigen::Vector3d(fieldPosition.translation().x(), fieldPosition.translation().y(), 0);
-                    Hfw.linear() = Eigen::AngleAxisd(Eigen::Rotation2Dd(fieldPosition.rotation()).angle(),
-                                                     Eigen::Vector3d::UnitZ())
-                                       .toRotationMatrix();
-
-                    Eigen::Affine3d Htf        = Htw * Hfw.inverse();
-                    Eigen::Vector3d kickTarget = Htf * Eigen::Vector3d(kickPlan.target.x(), kickPlan.target.y(), 0);
-
-                    // //approach point:
-                    Eigen::Vector2d ballToTarget = (kickTarget.head<2>() - position).normalized();
-                    Eigen::Vector2d kick_point   = position - ballToTarget * ball_approach_dist;
-
-                    if (position.norm() > slowdown_distance) {
-                        position = kick_point;
-                    }
-                    else {
-                        speedFactor   = slow_approach_factor;
-                        headingChange = std::atan2(ballToTarget.y(), ballToTarget.x());
-                        sideStep      = 1;
-                    }
-                }
+        // @brief - Save the plan cmd into latestCommand
+        on<Trigger<MotionCommand>, Sync<SimpleWalkPathPlanner>>().then(
+            [this](const MotionCommand& cmd) { latestCommand = cmd; });
 
+        //-------- UTILITY FUNCTIONS - Add extra behaviours for path planning here and in the switch case of this file
 
-                float angle = std::atan2(position.y(), position.x()) + headingChange;
+        void walkDirectly() {
+            std::unique_ptr<WalkCommand> command = std::make_unique<WalkCommand>(subsumptionId, latestCommand.walk_command);
+            emit(std::move(command));
+            emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {40, 11}}));
+        }
 
-                angle = std::min(turnSpeed, std::max(angle, -turnSpeed));
+        void determineSimpleWalkPath(const Ball& ball,
```

Comment:
Think this will mostly get replaced with a better path planner anyway, this PR was mostly just to make more sense of different path planning options/methods (strafe, to point, to ball, etc)